internal/operators/osc/operator.go

```diff
@@ -0,0 +1,178 @@
+		o.log.Error(message)
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message, err.Error()}}, err
+	}
+
+	cpu := requirements.CPUCores
+	if inventory.CPU.Count < cpu {
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{fmt.Sprintf("Insufficient CPU to deploy %s. Required CPU count is %d but found %d ", o.GetFullName(), cpu, inventory.CPU.Count)}}, nil
+	}
+
+	mem := requirements.RAMMib
+	memBytes := conversions.MibToBytes(mem)
+	if inventory.Memory.UsableBytes < memBytes {
+		usableMemory := conversions.BytesToMib(inventory.Memory.UsableBytes)
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{fmt.Sprintf("Insufficient memory to deploy %s. Required memory is %d MiB but found %d MiB", o.GetFullName(), mem, usableMemory)}}, nil
+	}
+
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetHostValidationID()}, nil
+}
+
+// GetHostRequirements provides operator's requirements towards the host
+func (o *operator) GetHostRequirements(ctx context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	log := logutil.FromContext(ctx, o.log)
+	preflightRequirements, err := o.GetPreflightRequirements(ctx, cluster)
+	if err != nil {
+		log.WithError(err).Errorf("Cannot retrieve preflight requirements for host %s", host.ID)
+		return nil, err
+	}
+
+	return preflightRequirements.Requirements.Master.Quantitative, nil
+}
+
+// GetPreflightRequirements returns operator hardware requirements that can be determined with cluster data only
+func (o *operator) GetPreflightRequirements(context context.Context, cluster *common.Cluster) (*models.OperatorHardwareRequirements, error) {
+	dependecies, err := o.GetDependencies(cluster)
+	if err != nil {
+		return &models.OperatorHardwareRequirements{}, err
+	}
+
+	return &models.OperatorHardwareRequirements{
+		OperatorName: o.GetName(),
+		Dependencies: dependecies,
+		Requirements: &models.HostTypeHardwareRequirementsWrapper{
+			Master: &models.HostTypeHardwareRequirements{
+				Qualitative: []string{
+					fmt.Sprintf("%d MiB of additional RAM", o.config.OscMemoryPerHostMiB),
+					fmt.Sprintf("%d additional CPUs", o.config.OscCPUPerHost),
+				},
+				Quantitative: &models.ClusterHostRequirementsDetails{
+					CPUCores: o.config.OscCPUPerHost,
+					RAMMib:   o.config.OscMemoryPerHostMiB,
+				},
+			},
+			Worker: &models.HostTypeHardwareRequirements{
+				Qualitative: []string{
+					fmt.Sprintf("%d MiB of additional RAM", o.config.OscMemoryPerHostMiB),
+					fmt.Sprintf("%d additional CPUs", o.config.OscCPUPerHost),
+				},
+				Quantitative: &models.ClusterHostRequirementsDetails{
+					CPUCores: o.config.OscCPUPerHost,
+					RAMMib:   o.config.OscMemoryPerHostMiB,
+				},
+			},
+		},
+	}, nil
```

Comment:
@xiangchunfu Here we should consider the SNO scenario. Let's discuss offline about the details.

Reply:
updated